src/vehicle.cpp

```diff
@@ -1666,77 +1761,110 @@ bool vehicle::split_vehicles( std::vector<std::vector <int>> new_vehs,
+            new_vehicle->cruise_velocity = cruise_velocity;
+            new_vehicle->cruise_on = cruise_on;
+            new_vehicle->engine_on = engine_on;
+            new_vehicle->tracking_on = tracking_on;
+            new_vehicle->camera_on = camera_on;
+        }
+        new_vehicle->last_fluid_check = last_fluid_check;
+
+        std::vector<player *> passengers;
         for( size_t new_part = 0; new_part < split_parts.size(); new_part++ ) {
             int mov_part = split_parts[ new_part ];
+            point cur_mount = parts[ mov_part ].mount;
+            point new_mount = cur_mount;
             player *passenger = nullptr;
             // Unboard any entities standing on any transferred part
             if( part_flag( mov_part, "BOARDABLE" ) ) {
-                std::vector<int> bp = boarded_parts();
-                for( auto elem : bp ) {
-                    if( elem == mov_part ) {
-                        passenger = get_passenger( mov_part );
-                        g->m.unboard_vehicle( passenger->pos() );
-                    }
+                passenger = get_passenger( mov_part );
+                if( passenger ) {
+                    passengers.push_back( passenger );
                 }
             }
 
-#if 0
-            // remove labels associated with the mov_part
-            std::string label_str;
-            const auto iter = labels.find( label( parts[ mov_part ].mount.x, parts[ mov_part ].mount.y ) );
-            const bool mv_label = iter != labels.end();
-            // Checking these twice to avoid calling the relatively expensive parts_at_relative() unnecessarily.
-            if( mv_label ) {
-                if( parts_at_relative( parts[ mov_part ].mount.x, parts[ mov_part ].mount.y, false ).empty() ) {
-                    if( mv_label ) {
-                        label_str = iter.begin().text;
-                        labels.erase( iter );
-                    }
-                }
-            }
-#endif
 
             // transfer the vehicle_part to the new vehicle
             new_vehicle->parts.emplace_back( parts[ mov_part ] );
-            if( split_mounts.empty() ) {
-                // calculate effective mount point
-                point new_mnt = parts[ mov_part ].mount - mnt_offset;
-                new_vehicle->parts.back().mount = rotate_mount( face.dir(), new_vehicle->face.dir(),
-                                                  point( 0, 0 ), new_mnt );
-            } else {
-                new_vehicle->parts.back().mount = split_mounts[ new_part ];
+            if( !split_mounts.empty() ) {
+                new_mount = split_mounts[ new_part ];
+                new_vehicle->parts.back().mount = new_mount;
             }
-            // put the passenger on the new vehicle
-            if( passenger ) {
-                g->m.board_vehicle( passenger->pos(), passenger );
+            // remove labels associated with the mov_part
+            std::string label_str;
```

Comment:
Why is this defined outside of the `if` block (it is only used inside of it)?

Reply:
Leftovers from previous versions of the code. Fixed, thanks!